### PR TITLE
Fixes bottom position bugs

### PR DESCRIPTION
**Pull request by alexey-m-ukolov**
Currently bottom position doesn't work as expected due to a couple of bugs:

1. **stickyStop** property doesn't take into account window height and because of that _stuck_ state applied only when element is above scroll top.

2. After initialization but before first scroll event target element receives `position: sticky` style but no `bottom: Xpx`. So it isn't visible at first. And I think that's fine. The problem is when we scrolled down and then back to the top (going from _sticky_ to _default_ state) element looses js-is-sticky class but not `bottom` style and does not receive js-is-stuck class. So browser with sticky support thinks that element is sticky at the moment (and it is), but there is no class to style it accordingly.  
We can either reset state to default (remove all classes and styles) or give element sticky state and all it's classes and styles. I chose the latter, since it makes more sense to me.

3. There is a typo which doesn't affect execution logic, but is misleading:
```
const isBottom = p.verticalPosition !== 'bottom'
```
Clearly if it is not equal to 'bottom' it is not bottom :)

---
I may be misunderstanding something (I don't get why we need _state_ property at all) and my fixes could break some other cases, but it works fine on my project with both top and bottom stickies and all test are green. Feel free to point out any mistakes.

### CODE REVIEW

**Comment by yowainwright** [comment]
@alexey-m-ukolov, thank you for your throughtful PR write-up and PR! 💪

I will look closer at this PR shortly. This is definitely a quick merge candidate.

File: src/stickybits.js [hunk 1]
```diff
@@ -217,7 +217,7 @@ class Stickybits {
     const el = it.el
     const parent = it.parent
     const isCustom = !this.isWin && p.positionVal === 'fixed'
-    const isBottom = p.verticalPosition !== 'bottom'
+    const isTop = p.verticalPosition !== 'bottom'
```
Comment (yowainwright) [inline src/stickybits.js]: 👌

File: src/stickybits.js [hunk 2]
```diff
@@ -226,11 +226,11 @@ class Stickybits {
...
       : el.offsetHeight
     it.offset = scrollElOffset + p.stickyBitStickyOffset
-    it.stickyStart = isBottom ? stickyStart - it.offset : 0
+    it.stickyStart = isTop ? stickyStart - it.offset : 0
     it.stickyChange = it.stickyStart + stickyChangeOffset
-    it.stickyStop = isBottom
+    it.stickyStop = isTop
       ? (stickyStart + parent.offsetHeight) - (it.el.offsetHeight + it.offset)
-      : stickyStart + parent.offsetHeight
+      : stickyStart + parent.offsetHeight - window.innerHeight
```
Comment (yowainwright) [inline src/stickybits.js]: Add parentheses...Maybe even a variable for this so it is more descriptive?

```javascript

it.stickyStop = isTop
      ? (stickyStart + parent.offsetHeight) - (it.el.offsetHeight + it.offset)	      
      : stickyStart + (parent.offsetHeight - window.innerHeight)

```

Something like:

```javascript

// for the ternaries; ~line 221
const calculatedParentScrollArea = parent.offsetHeight - window.innerHeight

// the for it.stickyStop; ~line 231
 it.stickyStop = isTop
      ? (stickyStart + parent.offsetHeight) - (it.el.offsetHeight + it.offset)	      
      : stickyStart + calculatedParentScrollArea

```

Feel free to come up with a better variable name.

Comment (alexey-m-ukolov) [reply src/stickybits.js]: Tbh, I don't think that extra variable will improve readability - it'll just increase cognitive load. As to parentheses - I've missed that it's in a project style and gladly add them.

**Comment by alexey-m-ukolov** [comment]
I've created variable after all, but for another value. Now it looks pretty good to me.
@yowainwright, please review.

P.S. I would've written it like this:
```
computeScrollOffsets (item) {
  const {props, el, parent} = item

  const isCustom = !this.isWin && props.positionVal === 'fixed'
  const isTop = props.verticalPosition !== 'bottom'

  const scrollElOffset = isCustom ? this.getTopPosition(props.scrollEl) : 0
  const stickyStart = isCustom
    ? this.getTopPosition(parent) - scrollElOffset
    : this.getTopPosition(parent)
  const stickyChangeOffset = props.customStickyChangeNumber !== null
    ? props.customStickyChangeNumber
    : el.offsetHeight
  const parentBottom = stickyStart + parent.offsetHeight

  item.offset = scrollElOffset + props.stickyBitStickyOffset
  item.stickyStart = isTop ? stickyStart - item.offset : 0
  item.stickyChange = item.stickyStart + stickyChangeOffset
  item.stickyStop = isTop
    ? parentBottom - (el.offsetHeight + item.offset)
    : parentBottom - window.innerHeight
}
```
But I see you like short variable names, so I won't submit a PR with that change. Unless you agree that it is indeed more readable :)

**Comment by yowainwright** [comment]
@alexey-m-ukolov, your destructuring does look nice and with gzipping, it may be smaller too. If you'd like to submit a pr with suggestions, I'm open to it.

**Pull request merged**